### PR TITLE
Fix protein chain selection (Issue #1421)

### LINKED ISSUE #1421: restraints: tests for `protein_chain_selection`
There are a few edge cases in that method that aren't being tested properly.

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #1421 

Specifically:
  - Selection no longer fails if no suitable chains were found
  - Adds tests for subselected chains and too small chains
  - Allows manually defining the minimum chain length for a protein chain.

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

File: openfe/protocols/restraint_utils/geometry/utils.py [hunk 1]
```diff
@@ -729,13 +730,16 @@ def protein_chain_selection(
...
+    at least ``min_chain_length`` (default: 30) residues which match the
+    ``protein`` selection of MDAnalysis.
 
     Parameters
     ----------
     atomgroup : mda.AtomgGroup
       The AtomGroup to select atoms from.
-    trim_chain_start: int
+    min_chain_length : int
+      The minimum number of residues in a protein chain. Default 30.
```
Comment (atravitz) [inline openfe/protocols/restraint_utils/geometry/utils.py]:
```suggestion
      The minimum number of residues to be considered a protein chain. Default 30.
```

Comment (atravitz) [reply openfe/protocols/restraint_utils/geometry/utils.py]: Also the Returns docstring should maybe include that it's returning the atoms of the *trimmed* protein chains, just to be precise.

``` 
 Returns
    -------
    atomgroup : mda.AtomGroup
      An AtomGroup containing all the atoms from the input AtomGroup
      which belong to the trimmed protein chains.
```

File: openfe/protocols/restraint_utils/geometry/utils.py [hunk 2]
```diff
@@ -729,13 +730,16 @@ def protein_chain_selection(
...
+    trim_chain_start : int
```
Comment (atravitz) [inline openfe/protocols/restraint_utils/geometry/utils.py:747]: I added a test for if the trimming is larger than the size of the residue. it returns an empty group.

File: openfe/tests/protocols/restraints/test_geometry_utils.py [hunk 3]
```diff
@@ -489,3 +489,30 @@ def test_protein_chain_selection(eg5_protein_ligand_universe):
...
+    """
+    Pass a subset of the residues in a chain
+    """
+
+    sele = protein_chain_selection(
+        atomgroup=eg5_pdb_universe.residues[:28].atoms,
+    )
+
+    assert len(sele.residues) == 18
+    assert len(sele) == len(eg5_pdb_universe.residues[10:28].atoms)
```
Comment (atravitz) [inline openfe/tests/protocols/restraints/test_geometry_utils.py]:
```suggestion
    assert len(sele) == 282
```
I think i'd prefer hard-coding this expected value.

**Review by atravitz — APPROVED** [review]
